src/components/Graph/Graph.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import useStore from '../../services/useStore';
+import { Chart, LineElement, LinearScale, CategoryScale, BarElement, PointElement } from 'chart.js';
+Chart.register(LineElement, LinearScale, CategoryScale, BarElement, PointElement);
+import { Line } from 'react-chartjs-2';
+import { format } from 'date-fns';
+import { orderBy } from 'lodash';
+
+export default function Graph() {
+	const tasks = useStore(state => state.tasks);
+
+	const pastDates = tasks.flatMap(task => [task.startDate, task.doneDate]);
+	const cleanPastDates = pastDates.filter(date => date !== undefined);
+	const orderedPastDates = orderBy(cleanPastDates);
+
+	const tasksCountAtDate = orderedPastDates.map(
+		date => tasks.filter(task => task.startDate < date).length
+	);
+	const shiftTasksCountAtDate = tasksCountAtDate.map(count => count - 1);
+	const TasksDoneAtDate = orderedPastDates.map(
+		date => tasks.filter(task => task.doneDate < date).length
+	);
+	const substractTasksDone = shiftTasksCountAtDate.map(
+		(task, index) => task - TasksDoneAtDate[index]
```

Comment:
This is ok, but the naming could be a bit clearer here. I guess the map here gets taskCount (a number) and not a task (an object).

Reply:
True